### PR TITLE
Treat identities as equivalent in `pauli.are_identical_pauli_words`

### LINKED ISSUE #4155: [BUG] Wrong expectation value of Hamiltonian with return qml.expval(H)
### Expected behavior

Expected to get the same expectation value of a qml.Hamiltonian no matter
1. how the operators and  the corresponding coefficient are ordered when it was constructed.
2. applied the grouping method ('QWC') or not.

### Actual behavior

When the below conditions are satisfied, the expectation value will not take the value (coefficient) of qml.Identity into account.
1. There is a "grouping_type" argument when qml.Hamiltonian is constructed.
2. The operator list is not  in the order of qubit number.  

Example
`H_disordered = qml.Hamiltonian([1.0, 1.0, 1.0], [qml.PauliZ(1), qml.PauliZ(0), qml.Identity(0)], grouping_type='QWC')`
The list of operator is disordered in terms of qubit number.
However its expectation value differs with the value of qml.Identity from an ordered Hamiltonian  
`H_ordered = qml.Hamiltonian([1.0, 1.0, 1.0], [qml.Identity(0), qml.PauliZ(0), qml.PauliZ(1)], grouping_type='QWC')`

### Additional information

_No response_

### Source code

```shell
import pennylane as qml

dev = qml.device("default.qubit", wires=2, shots=1000)
@qml.qnode(dev)
def circuit():
    return qml.expval(H)

H = qml.Hamiltonian([1.0, 1.0, 1.0], [qml.PauliZ(0), qml.PauliZ(1), qml.Identity(0)], grouping_type='QWC')
print(circuit())

H = qml.Hamiltonian([1.0, 1.0, 1.0], [qml.PauliZ(1), qml.PauliZ(0), qml.Identity(0)], grouping_type='QWC')
print(circuit())

H = qml.Hamiltonian([1.0, 1.0, 1.0], [qml.PauliZ(1), qml.PauliZ(0), qml.Identity(0)], grouping_type='commuting')
print(circuit())

H = qml.Hamiltonian([1.0, 1.0, 1.0], [qml.PauliZ(1), qml.PauliZ(0), qml.Identity(0)])
print(circuit())
```

### Tracebacks

_No response_

### System information

```shell
Platform info:           macOS-13.0.1-arm64-arm-64bit
Python version:          3.9.12
Numpy version:           1.23.3
Scipy version:           1.9.3
Installed devices:
- default.gaussian (PennyLane-0.30.0)
- default.mixed (PennyLane-0.30.0)
- default.qubit (PennyLane-0.30.0)
- default.qubit.autograd (PennyLane-0.30.0)
- default.qubit.jax (PennyLane-0.30.0)
- default.qubit.tf (PennyLane-0.30.0)
- default.qubit.torch (PennyLane-0.30.0)
- default.qutrit (PennyLane-0.30.0)
- null.qubit (PennyLane-0.30.0)
- lightning.qubit (PennyLane-Lightning-0.30.0)
```

### Existing GitHub issues

- [X] I have searched existing GitHub issues to make sure the issue does not already exist.

### PR DESCRIPTION
**Pull request by albi3ro**
Fixes #4155 

```
>>> binary_repr = qml.pauli.pauli_to_binary(qml.Identity("b"), wire_map={"a": 0, "b": 1})
>>> qml.pauli.binary_to_pauli(binary_repr, wire_map={"a": 0, "b": 1})
Identity(wires=['a'])
```

Converting an `Identity` to a binary form and back removes the wires it was originally on.  This causes problems when computing the grouping indices on a Hamiltonian because the original operation and the returned operation are no longer the same.

This problem can be circumvented by making all identities identical:
```
>>> qml.pauli.are_identical_pauli_words(qml.Identity("a"), qml.Identity("b"))
True
```

Given the two terms do inherently represent the same thing, this decision should be fine.

### CODE REVIEW

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -168,6 +168,9 @@
 * Removes a patch in `interfaces/autograd.py` that checks for the `strawberryfields.gbs` device.  That device
   is pinned to PennyLane <= v0.29.0, so that patch is no longer necessary.
 
+* `qml.pauli.are_identical_pauli_words` now treats all identities as equal. Identity terms on Hamiltonians with non-standard
+  wire orders are no longer eliminated.
```
Comment (albi3ro) [inline doc/releases/changelog-dev.md:172]:
```suggestion
  wire orders are no longer eliminated.
  [(#4161)](https://github.com/PennyLaneAI/pennylane/pull/4161)
```

**Review by timmysilv — APPROVED** [review]
very clean fix, nice! I can't help but ask about the original concern you mention in the PR description when going Identity -> binary_repr -> Identity, which will still occur after this fix. The first step is lossy, so I'm good to have this unexpected behaviour because what else can you do. However, it makes me wonder if we're missing little bugs like this elsewhere. I suppose it's hard to know until we hit them. Happy to approve regardless, but something to consider

**Review by eddddddy — APPROVED** [review]
Thanks for this @albi3ro 💯

File: doc/releases/changelog-dev.md [hunk 2]
```diff
@@ -168,6 +168,10 @@
...
+  [(#4161)](https://github.com/PennyLaneAI/pennylane/pull/4161)
```
[the remark this answers is not in the document]
Comment (albi3ro) [reply doc/releases/changelog-dev.md:173]: Given we didn't have any tests for this sort of behaviour, I'm not too concerned.

[the remark this answers is not in the document]
Comment (eddddddy) [reply doc/releases/changelog-dev.md:173]: Fair enough.

**Comment by albi3ro** [comment]
> very clean fix, nice! I can't help but ask about the original concern you mention in the PR description when going Identity -> binary_repr -> Identity, which will still occur after this fix. The first step is lossy, so I'm good to have this unexpected behaviour because what else can you do. However, it makes me wonder if we're missing little bugs like this elsewhere. I suppose it's hard to know until we hit them. Happy to approve regardless, but something to consider

Once we port all these "utilities" over to relying on `PauliWord` and `PauliSentence` instead, we won't have to deal with the binary rep quirks anymore, so I'm not too concerned about the longevity of this problem.

**Pull request merged**